tests/unit/db/index_tests.py

```diff
@@ -0,0 +1,588 @@
+                 'language': 'query',
+                 'indexes': {'index001': 
+                     {'index': {'index_array_lengths': True,
+                                'fields': [{'name': 'name', 'type': 'string'},
+                                           {'name': 'age', 'type': 'number'}],
+                                'default_field': {'enabled': True,
+                                                  'analyzer': 'german'},
+                                'default_analyzer': 'keyword',
+                                'selector': {}},
+                      'analyzer': {'name': 'perfield',
+                                   'default': 'keyword',
+                                   'fields': {'$default': 'german'}}}}}
+            )
+
+    def test_create_a_search_index_invalid_argument(self):
+        """
+        Test that a TEXT index is not created when an invalid argument is given.
+        """
+        index = SearchIndex(self.db, 'ddoc001', 'index001', foo='bar')
+        with self.assertRaises(CloudantArgumentError) as cm:
+            index.create()
+        err = cm.exception
+        self.assertEqual(str(err), 'Invalid argument: foo')
+
+    def test_create_a_search_index_invalid_fields_value(self):
+        """
+        Test that a TEXT index is not created when an invalid fields value is
+        given.
+        """
+        index = SearchIndex(self.db, 'ddoc001', 'index001', fields=5)
+        with self.assertRaises(CloudantArgumentError) as cm:
+            index.create()
+        err = cm.exception
+        self.assertEqual(
+            str(err),
+            'Argument fields is not an instance of expected type: '
+            '<type \'list\'>'
+        )
+
+    def test_create_a_search_index_invalid_default_field_value(self):
+        """
+        Test that a TEXT index is not created when an invalid default_field
+        value is given.
+        """
+        index = SearchIndex(self.db, 'ddoc001', 'index001', default_field=5)
+        with self.assertRaises(CloudantArgumentError) as cm:
+            index.create()
+        err = cm.exception
+        self.assertEqual(
+            str(err),
+            'Argument default_field is not an instance of expected type: '
+            '<type \'dict\'>'
+        )
+
+    def test_create_a_search_index_invalid_selector_value(self):
+        """
+        Test that a TEXT index is not created when an invalid default_field
+        value is given.
+        """
+        index = SearchIndex(self.db, 'ddoc001', 'index001', selector=5)
+        with self.assertRaises(CloudantArgumentError) as cm:
+            index.create()
+        err = cm.exception
+        self.assertEqual(
+            str(err),
+            'Argument selector is not an instance of expected type: '
+            '<type \'dict\'>'
+        )
+
+    def test_text_index_via_query(self):
+        """
+        Test that a created index will produce expected query results.
```

Comment:
Minor, but this could say `SearchIndex` instead of `index` to more clearly differentiate it from the test for `Index` above.